contracts/AirDrop.sol

```diff
@@ -0,0 +1,252 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.7;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/utils/cryptography/draft-EIP712.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "../interfaces/IAirDrop.sol";
+
+/**
+ * @title AirDrop
+ * @author ZaniInc
+ * @notice This smart contract is for send free tokens to loyal
+ * members of crypto-project
+ */
+contract AirDrop is EIP712, IAirDrop, Ownable {
+    using SafeERC20 for IERC20;
+    using Address for address;
+
+    /**
+     * @dev Store balances of members
+     * @return amounts of available tokens & ether
+     */
+    mapping(address => uint256) public balanceOfTokens;
+    mapping(address => uint256) public balanceOfEther;
+
+    /**
+     * @dev Use for sign message with EIP712 standart
+     * @notice 'DROP_HASH' is signature of primaryType 'Drop'
+     */
+    bytes32 public constant DROP_HASH =
+        keccak256(
+            "Drop(uint256[] amount,uint256 deadline,address[] recepient,bool ether)"
+        );
+
+    IERC20 public token;
+
+    /**
+     * @dev Set 'token' IERC20 to interact with thrid party token
+     *
+     * Put address to state variable '_owner'
+     *
+     * @param token_ - of ERC20 contract
+     */
+    constructor(address token_) EIP712("AirDrop", "1") {
+        Address.isContract(token_);
```

Comment:
Address.isContract return true/false , you need check return variable.
also use "token_.isContract()" because you declarate in top use Address for address.